### PR TITLE
feat: add cleavr preset

### LINKED ISSUE #522: feat: add Cleavr.io preset
Add Cleavr.io preset  for Nitro. 

https://cleavr.io

### PR DESCRIPTION
**Pull request by anishghimire862**
### 🔗 Linked issue

#522 

### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Add Cleavr.io (https://cleavr.io) preset  for Nitro.

Resolves #522 

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by pi0** [comment]
Thanks for the PR. Do you know if clever is providing any distinct environment variable to auto-detect the preset? it can be added [here](https://github.com/unjs/std-env/blob/main/src/providers.ts).

**Comment by anishghimire862** [comment]
Hello @pi0,

Right now we pass `node-server` as preset. We plan to pass `cleavr` as preset as soon as this PR gets merged.

File: docs/content/2.deploy/providers/cleavr.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+---
+title: Cleavr
+description: 'Discover Cleavr preset for Nitro!'
+---
+
+## Cleavr
+
+**Preset:** `cleavr` ([switch to this preset](/deploy/#changing-the-deployment-preset))
+
+Nitro provides a built-in preset to generate output format compatible with [Cleavr.io](https://www.cleavr.io/).
```
Comment (pi0) [inline docs/content/2.deploy/providers/cleavr.md:10]: www. subdomains seems broken. It redirects me to https://country-lookup.cleaver.rocks/

**Comment by pi0** [comment]
> We plan to pass cleavr as preset as soon as this PR gets merged.

This is nice but I'm wondering if we can make preset smarter to work out of the box without configuration. Most of cloud provider's provide a unique environment variable (like CLEAVER=1) in their CI/CI so that nitro can auto-configure the preset.

**Comment by anishghimire862** [comment]
Closing the PR for now. We're working on an improvement to accept an environment variable to auto-configure the preset. 

I'll re-open the PR soon.

Thank you.

**Comment by pi0** [comment]
No worries. I've reopened your PR to keep track. If it takes more efforts, I'm also okay to list and iterate to make it auto detected preset.

**Comment by anishghimire862** [comment]
I think we misunderstood somehow. I've added Cleavr to auto detectable providers and sent a PR.

When running the build we'll pass an environment variable CLEAVR=1

This is what the build command will look like: 

`CLEAVR=1 npm run build --production`

This is how the PM2 Ecosystem config will look like containing env: 

module.exports = {
  ...,
  env: {  
    ...,
    CLEAVR=1
  }
}

I'm happy to make relevant changes if you think this doesn't work.

**Comment by anishghimire862** [comment]
Hello @pi0,

The other PR for auto-detectable preset got merged. Is there any more changes that I should make here?

**Review by pi0 — APPROVED** [review]

**Comment by pi0** [comment]
Thanks @anishghimire862 and sorry it was delayed.

I have released `std-env` but could not push to your fork. Will add auto detect in main.

**Comment by pi0** [comment]
Added via https://github.com/unjs/nitro/commit/d5bbdeb8411695a7e97f90284afe3d6fe43e9375

**Pull request merged**